src/tools/weave/KeyUtils.cpp

```diff
@@ -59,6 +59,24 @@ bool ReadPrivateKey(const char *fileName, const char *prompt, EVP_PKEY *& key)
     return res;
 }
 
+bool ReadPublicKey(const char *fileName, EVP_PKEY *& key)
```

Comment:
Identical code (except for call to potentially refactorable `DecodePublicKey`) to ReadPrivateKey.  Opportunity for refactoring?